File: ljl-blog-server/src/main/java/com/ljlblogserver/entity/Article.java
package com.ljlblogserver.entity;

import lombok.Data;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Data
public class Article {

    private Long id;
    private String articleType;
    private String slug;
    private String title;
    private String summary;
    private String coverPath;
    private String contentPath;
    private Long categoryId;
    private String categoryName;
    private String categorySlug;
    private LocalDate publishDate;
    private Integer readTime;
    private Boolean featured;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    private List<String> tags = new ArrayList<>();
}

File: ljl-blog-web/src/types/article.ts
/** 文章 / 学习文档通用结构 */
export interface Article {
  id: string
  slug: string
  title: string
  summary: string
  content?: string
  cover?: string
  category: string
  categorySlug: string
  tags: string[]
  publishDate: string
  readTime?: number
  featured?: boolean
}
